### PR TITLE
Add customized static cache implementation

### PR DESCRIPTION
**Pull request by helunwencser**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #4478
* #4491
* __->__ #4490

Differential Revision: [D60554455](https://our.internmc.facebook.com/intern/diff/D60554455)

### CODE REVIEW

**Comment by helunwencser** [comment]
@helunwencser has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D60554455).

File: examples/models/phi-3-mini/static_cache.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+# LICENSE file in the root directory of this source tree.
+
+
+from typing import Optional
+
+import torch
+from transformers import PretrainedConfig, StaticCache
+
+
+class ETStaticCache(StaticCache):
```
Comment (iseeyuan) [inline examples/models/phi-3-mini/static_cache.py:14]: Could you add comments on why this custom static cache class is needed?

Comment (helunwencser) [reply examples/models/phi-3-mini/static_cache.py:14]: of course

**Review by iseeyuan — APPROVED** [review]
LGTM. Thanks!

**Pull request closed without merging**